### PR TITLE
feat(stark-demo): add the ngx form error to the showcase

### PR DESCRIPTION
**Pull request by nicanac**
ISSUES CLOSED: #1195

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
Add the ngx form error to the showcase

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number:#1195

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other information

### CODE REVIEW

File: showcase/src/app/demo-ui/pages/reactive-forms-with-ngx-form-errors/demo-reactive-forms-with-ngx-form-errors-page.component.scss [hunk 1]
```diff
@@ -0,0 +1,90 @@
+$monitor-query: "screen and (min-width:1200px)";
```
Comment (SuperITMan) [inline showcase/src/app/demo-ui/pages/reactive-forms-with-ngx-form-errors/demo-reactive-forms-with-ngx-form-errors-page.component.scss]: Could you please use variables from "stark-ui" instead ? 😊 
```css
@import "~@nationalbankbelgium/stark-ui/assets/styles/media-queries";
```

Then you can use the variables `$desktop-screen-query`, `$tablet-only-query` and `$mobile-only-query`.

File: showcase/src/app/demo-ui/pages/reactive-forms-with-ngx-form-errors/demo-reactive-forms-with-ngx-form-errors-page.component.scss [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+$table-query: "screen and (max-width: 1200px)";
+$mobile-query: "screen and (max-width: 600px)";
+button {
+  margin: 8px;
+}
+hr.ngx-form-demo-line {
```
Comment (SuperITMan) [inline showcase/src/app/demo-ui/pages/reactive-forms-with-ngx-form-errors/demo-reactive-forms-with-ngx-form-errors-page.component.scss]: This is not used in your template. Could you please remove it ? 😊

File: showcase/src/app/demo-ui/pages/reactive-forms-with-ngx-form-errors/demo-reactive-forms-with-ngx-form-errors-page.component.scss [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  margin-top: 32px;
+  margin-bottom: 28px;
+}
+div.mat-form-field-infix {
```
Comment (SuperITMan) [inline showcase/src/app/demo-ui/pages/reactive-forms-with-ngx-form-errors/demo-reactive-forms-with-ngx-form-errors-page.component.scss]: This is not used in the template either.

File: showcase/src/app/demo-ui/components/card/card.component.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import { Component, HostBinding, Input } from "@angular/core";
+
+@Component({
+	selector: "app-card",
+	templateUrl: "./card.component.html",
+	styleUrls: ["./card.component.scss"]
+})
+export class CardComponent {
```
Comment (SuperITMan) [inline showcase/src/app/demo-ui/components/card/card.component.ts]: Maybe this component could be part of "reactive-forms-with-ngx-form-errors" folder as is this the case for "generic-search" 😊

File: showcase/src/app/app-menu.config.ts [hunk 5]
```diff
@@ -242,6 +242,21 @@ export const APP_MENU_CONFIG: StarkMenuConfig = {
 					label: "Services",
 					isVisible: true,
 					isEnabled: true
+				},
+				{
+					id: "menu-stark-ui-forms",
+					label: "Forms",
```
[the remark this answers is not in the document]
Comment (christophercr) [reply showcase/src/app/app-menu.config.ts]: I would propose to move it to the "StyleGuide" cause it is more like a solution we propose for error messages. The other option would be to put it in a completely separate section in the menu... but I'm not sure...

File: showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html [hunk 6]
```diff
@@ -0,0 +1,141 @@
+<form [formGroup]="formGroup" (ngSubmit)="onSubmitUserDetails(formGroup)" class="form-group">
+	<div class="grid">
+		<app-card class="form-card">
+			<mat-card-content>
+				<mat-form-field class="full-width">
+					<input matInput [placeholder]="'SHOWCASE.FORMS.NGX_FORMS.USER_NAME' | translate" formControlName="username" required />
```
Comment (christophercr) [inline showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html]: Can you replace this translation key by the translated text?

File: showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+						<ng-template ngxFormErrors="username" #usernameField="ngxFormErrors"></ng-template>
+					</mat-error>
+				</mat-form-field>
+
+				<div formGroupName="matchingPasswords" ngxFormErrorsGroup="passwords-group">
+					<mat-form-field class="full-width">
+						<input
+							matInput
+							type="password"
+							[placeholder]="'SHOWCASE.FORMS.NGX_FORMS.PASSWORD' | translate"
```
Comment (christophercr) [inline showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html]: Can you replace this translation key by the translated text?

File: showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+								#passwordField="ngxFormErrors"
+							></ng-template>
+						</mat-error>
+					</mat-form-field>
+
+					<mat-form-field class="full-width">
+						<input
+							matInput
+							type="password"
+							[placeholder]="'SHOWCASE.FORMS.NGX_FORMS.CONFIRM_PASSWORD' | translate"
```
Comment (christophercr) [inline showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html]: Can you replace this translation key by the translated text?

File: showcase/src/app/app-menu.config.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (nicanac) [reply showcase/src/app/app-menu.config.ts]: For now it has been moved in the welcome part.
Nevertheless, I can still move it.

File: showcase/src/app/app-menu.config.ts [hunk 9]
```diff
@@ -242,6 +242,21 @@ export const APP_MENU_CONFIG: StarkMenuConfig = {
...
+					isVisible: true,
+					isEnabled: true,
+					entries: [
+						{
+							id: "ngx-form-errors",
+							label: "Reactive Forms with NgxFormErros",
```
[the remark this answers is not in the document]
Comment (nicanac) [reply showcase/src/app/app-menu.config.ts]: Reactive Form errors sounds good to me

File: showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.scss [hunk 10]
```diff
@@ -0,0 +1,90 @@
+$monitor-query: "screen and (min-width:1200px)";
+$table-query: "screen and (max-width: 1200px)";
+$mobile-query: "screen and (max-width: 600px)";
```
Comment (christophercr) [inline showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.scss]: Can you change these 3 lines too to keep them in sync with the source file `welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.scss`

In fact, can you update this file entirely?

File: showcase/src/app/welcome/pages/reactive-form-errors/components/card/card.component.ts [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+	public primaryColor!: boolean;
+	@HostBinding("class.app-color-accent")
+	public accentColor!: boolean;
+	@HostBinding("class.app-color-warning")
+	public warningColor!: boolean;
+	@HostBinding("class.app-color-success")
+	public successColor!: boolean;
+
+	@Input()
+	/* tslint:disable-next-line:max-union-size */
```
Comment (christophercr) [inline showcase/src/app/welcome/pages/reactive-form-errors/components/card/card.component.ts]: This TSLint violation is really easy to solve, you can create an alias type to contain the different colors. Could you fix this? That way we keep less things on our backlog ;)

File: showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.html [hunk 12]
```diff
@@ -0,0 +1,163 @@
+<div class="container">
+	<h1 class="mat-display-1" translate>
+		SHOWCASE.NGX_FORM_ERRORS.TITLE
+	</h1>
+	<h1 translate>SHOWCASE.DEMO.SHARED.EXAMPLE_VIEWER_LIST</h1>
+	<section class="stark-section">
+		<example-viewer
+			class="progress-indicator-example"
+			[extensions]="['HTML', 'SCSS', 'TS']"
+			filesPath="reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors"
```
Comment (christophercr) [inline showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.html]: Can you rename the folder of the examples to `reactive-form-errors` to keep consistency?

File: showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html (hunk 6 above)
Comment (christophercr) [inline showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html]: Can you replace this translation key by the translated text?

File: showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html (hunk 7 above)
Comment (christophercr) [inline showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html]: Can you replace this translation key by the translated text?

File: showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+							matInput
+							type="password"
+							[placeholder]="'SHOWCASE.FORMS.NGX_FORMS.PASSWORD' | translate"
+							formControlName="password"
+							required
+						/>
+						<mat-error>
+							<ng-template
+								ngxFormErrors="matchingPasswords.password"
+								ngxFormErrorsFieldName="SHOWCASE.FORMS.NGX_FORMS.PASSWORD_ALIAS"
```
Comment (christophercr) [inline showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html]: Can you replace this translation key by the translated text?

File: showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html (hunk 8 above)
Comment (christophercr) [inline showcase/src/assets/examples/reactive-forms-with-ngx-form-errors/reactive-forms-with-ngx-form-errors.html]: Can you replace this translation key by the translated text?

File: showcase/src/assets/translations/en.json [hunk 14]
```diff
@@ -393,6 +393,16 @@
         }
       }
     },
+    "FORMS": {
+      "NGX_FORMS": {
```
Comment (christophercr) [inline showcase/src/assets/translations/en.json]: Why is this block apart? Shouldn't it be in the the block of `NGX_FORM_ERRORS`? Moreover, I think it should be `REACTIVE_FORM_ERRORS` instead, so it is in sync with the actual page

File: showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.html [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+							<li [hidden]="!confirmPasswordField.hasErrors">
+								<ng-template ngxFormErrors="matchingPasswords.confirmPassword"></ng-template>
+							</li>
+						</ul>
+					</app-card>
+				</div>
+			</form>
+		</example-viewer>
+	</section>
+</div>
```
Comment (christophercr) [inline showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.html]: ⚠️ don't forget to add the "Reference" links to the NGX Form errors repo... like we do with the UI components for example );

**Comment by christophercr** [comment]
@nicanac thinking a bit more on this, in fact it is not necessary to add the other section to show the configuration needed for the `NgxFormErrorsModule` and the `entryComponents` that I mentioned before... normally that is already explained in the docs of the `ngx-form-errors` package... so just adding the Reference link to those should be enough... what do you think?

That way it is safer as we don't need to duplicate documentation...

**Comment by nicanac** [comment]
> @nicanac thinking a bit more on this, in fact it is not necessary to add the other section to show the configuration needed for the `NgxFormErrorsModule` and the `entryComponents` that I mentioned before... normally that is already explained in the docs of the `ngx-form-errors` package... so just adding the Reference link in the demo page should be enough... what do you think?
> 
> That way it is safer as we don't need to duplicate documentation...

@christophercr , I do agree, it's the reason why I didn't include it at first.
I'll add the reference link to the demo page, and we should be fine.

File: showcase/src/app/app-menu.config.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (nicanac) [reply showcase/src/app/app-menu.config.ts]: I propose, we keep it in the welcome part for this release. And move it in a separated issue.
What do you think ?

File: showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html [hunk 16]
```diff
@@ -0,0 +1,141 @@
+<form [formGroup]="formGroup" (ngSubmit)="onSubmitUserDetails(formGroup)" class="form-group">
+	<div class="grid">
+		<app-card class="form-card">
+			<mat-card-content>
+				<mat-form-field class="full-width">
+					<input matInput [placeholder]="'SHOWCASE.FORMS.NGX_FORMS.USER_NAME' | translate" formControlName="username" required />
```
Comment (christophercr) [inline showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html]: Can you replace this translation key by the translated text?

File: showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+						<ng-template ngxFormErrors="username" #usernameField="ngxFormErrors"></ng-template>
+					</mat-error>
+				</mat-form-field>
+
+				<div formGroupName="matchingPasswords" ngxFormErrorsGroup="passwords-group">
+					<mat-form-field class="full-width">
+						<input
+							matInput
+							type="password"
+							[placeholder]="'SHOWCASE.FORMS.NGX_FORMS.PASSWORD' | translate"
```
Comment (christophercr) [inline showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html]: Can you replace this translation key by the translated text?

File: showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+								#passwordField="ngxFormErrors"
+							></ng-template>
+						</mat-error>
+					</mat-form-field>
+
+					<mat-form-field class="full-width">
+						<input
+							matInput
+							type="password"
+							[placeholder]="'SHOWCASE.FORMS.NGX_FORMS.CONFIRM_PASSWORD' | translate"
```
Comment (christophercr) [inline showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html]: Can you replace this translation key by the translated text?

File: showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+							matInput
+							type="password"
+							[placeholder]="'SHOWCASE.FORMS.NGX_FORMS.PASSWORD' | translate"
+							formControlName="password"
+							required
+						/>
+						<mat-error>
+							<ng-template
+								ngxFormErrors="matchingPasswords.password"
+								ngxFormErrorsFieldName="SHOWCASE.FORMS.NGX_FORMS.PASSWORD_ALIAS"
```
Comment (christophercr) [inline showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html]: Can you replace this translation key by the translated text?

File: showcase/src/assets/translations/fr.json [hunk 20]
```diff
@@ -393,6 +393,16 @@
         }
       }
     },
+    "FORMS": {
+      "NGX_FORMS": {
```
Comment (christophercr) [inline showcase/src/assets/translations/fr.json]: Why is this block apart? Shouldn't it be in the the block of NGX_FORM_ERRORS? Moreover, I think it should be REACTIVE_FORM_ERRORS instead, so it is in sync with the actual page

File: showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.html [hunk 21]
```diff
@@ -0,0 +1,163 @@
...
+
+							<li [hidden]="!confirmPasswordField.hasErrors">
+								<ng-template ngxFormErrors="matchingPasswords.confirmPassword"></ng-template>
+							</li>
+						</ul>
+					</app-card>
+				</div>
+			</form>
+		</example-viewer>
+	</section>
```
Comment (christophercr) [inline showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.html]: ⚠️ don't forget to add the "Reference" links to the NGX Form errors repo... like we do with the UI components for example );

File: showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html (hunk 16 above)
Comment (nicanac) [reply showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html]: Sorry, I think I don't get it :-(
You mean, I shouldn't use the translation here ?

File: showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html (hunk 17 above)
Comment (nicanac) [reply showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html]: Sorry, I think I don't get it :-(
You mean, I shouldn't use the translation here ?

File: showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html (hunk 19 above)
Comment (nicanac) [reply showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html]: Sorry, I think I don't get it :-(
You mean, I shouldn't use the translation here ?

File: showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html (hunk 18 above)
Comment (nicanac) [reply showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html]: Sorry, I think I don't get it :-(
You mean, I shouldn't use the translation here ?

File: showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html (hunk 16 above)
Comment (nicanac) [reply showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html]: Sorry, I think I don't get it :-(
You mean, I shouldn't use the translation here ?

File: showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html (hunk 19 above)
Comment (christophercr) [reply showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html]: yep, exactly, in the example files we should only use the actual text... the translation keys are not relevant in these

Comment (nicanac) [reply showcase/src/assets/examples/reactive-forms-errors/reactive-forms-errors.html]: Okok, I get it now :-)

File: showcase/src/assets/translations/fr.json (hunk 20 above)
Comment (christophercr) [inline showcase/src/assets/translations/fr.json]: Why is this block apart? Shouldn't it be in the the block of `NGX_FORM_ERRORS`? Moreover, I think it should be `REACTIVE_FORM_ERRORS` instead, so it is in sync with the actual page

File: showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.ts [hunk 22]
```diff
@@ -0,0 +1,69 @@
...
+	selector: "showcase-reactive-forms-with-ngx-form-errors",
+	templateUrl: "./demo-reactive-form-errors-page.component.html",
+	styleUrls: ["./demo-reactive-form-errors-page.component.scss"]
+})
+export class DemoReactiveFormErrorsPageComponent implements OnInit {
+	public collapsed: boolean[] = [false, false, true];
+
+	public referenceList: ReferenceLink[] = [
+		{
+			label: "NGX Form errors repository",
```
Comment (christophercr) [inline showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.ts]: I think it is better to use the word `library` instead of `repository` so it is generic enough cause maybe we will point to a different page in the future (docs or demo page, who knows) ;)

File: showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.ts [hunk 23]
```diff
@@ -0,0 +1,69 @@
...
+			url: "https://www.npmjs.com/package/@nationalbankbelgium/ngx-form-errors"
```
Comment (christophercr) [inline showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.ts]: Question: shouldn't we point to the GitHub page? In my case I prefer to go to the GitHub page of a package because there I can find all the relevant info: releases, changelog and code examples. Normally in NPM you can just find the versions and the README but no changelog so you'll have to go to the GitHub repo for that :s

@SuperItMan @carlo-nomes  What do you guys think?

Comment (nicanac) [reply showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.ts]: We could add the both ?

File: showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.ts (hunk 22 above)
Comment (nicanac) [reply showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.ts]: Ok, so I can phrase it this way : Reactive form errors library ?

Comment (christophercr) [reply showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.ts]: I would say: `NGX Form erros library` 😉

Comment (nicanac) [reply showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.ts]: okok

File: showcase/src/assets/translations/fr.json (hunk 20 above)
Comment (christophercr) [inline showcase/src/assets/translations/fr.json]: Why is this block apart? Shouldn't it be in the the block of `NGX_FORM_ERRORS`? Moreover, I think it should be `REACTIVE_FORM_ERRORS` instead, so it is in sync with the actual page

File: showcase/src/app/demo-ui/demo-ui.module.ts [hunk 24]
```diff
@@ -81,6 +83,9 @@ import {
 		UIRouterModule.forChild({
 			states: DEMO_STATES
 		}),
+		CommonModule,
+		FormsModule,
+		ReactiveFormsModule,
```
Comment (carlo-nomes) [inline showcase/src/app/demo-ui/demo-ui.module.ts]: These are already imported through the shared module. 😉

File: showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.scss [hunk 25]
```diff
@@ -0,0 +1,80 @@
+@import "~@nationalbankbelgium/stark-ui/assets/styles/media-queries";
+button {
+  margin: 8px;
+}
+.grid {
```
Comment (carlo-nomes) [inline showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.scss]: grid won't work in IE, maybe you could use flexbox instead

File: showcase/src/app/welcome/pages/reactive-form-errors/parent-error-state-matcher.ts [hunk 26]
```diff
@@ -0,0 +1,12 @@
+import { ErrorStateMatcher } from "@angular/material/core";
+import { FormControl, FormGroupDirective, NgForm } from "@angular/forms";
+
+/** Error when invalid control is dirty, touched, or submitted. */
```
Comment (carlo-nomes) [inline showcase/src/app/welcome/pages/reactive-form-errors/parent-error-state-matcher.ts]: Could you add this on a new line? EG:
```JS
/**
 *  Error when invalid control is dirty, touched, or submitted.
 */
```

File: showcase/src/app/welcome/pages/reactive-form-errors/password-validator.ts [hunk 27]
```diff
@@ -0,0 +1,32 @@
...
+		let value: string | undefined;
+		let valid = true;
+		for (const key in formGroup.controls) {
+			if (formGroup.controls.hasOwnProperty(key)) {
+				const control: FormControl = <FormControl>formGroup.controls[key];
+
+				if (value === undefined) {
+					value = control.value;
+				} else {
+					if (value !== control.value) {
```
Comment (carlo-nomes) [inline showcase/src/app/welcome/pages/reactive-form-errors/password-validator.ts]: could you merge this in a else if?

File: showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.ts [hunk 28]
```diff
@@ -0,0 +1,69 @@
...
+	public parentErrorStateMatcher: ErrorStateMatcher = new ParentErrorStateMatcher();
+	public passwordPattern = "^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])[a-zA-Z0-9]+$";
+	public showValidationDetails = false;
+	public showValidationSummary = true;
+	public constructor(private formBuilder: FormBuilder, @Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService) {}
+
+	public ngOnInit(): void {
+		this.formGroup = this.formBuilder.group({
+			username: [undefined, Validators.required],
+			matchingPasswords: new FormGroup(
```
Comment (carlo-nomes) [inline showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.ts]: could you use the formbuilder for the group and form controls?

File: showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.ts [hunk 29]
```diff
@@ -0,0 +1,69 @@
...
+							Validators.maxLength(10),
+							Validators.required,
+							Validators.pattern(this.passwordPattern) // this is for the letters (both uppercase and lowercase) and numbers validation
+						])
+					),
+					confirmPassword: new FormControl("", Validators.required)
+				},
+				{
+					validators: (formGroup: AbstractControl): any => {
+						return PasswordValidator.areEqual(<FormGroup>formGroup);
```
Comment (carlo-nomes) [inline showcase/src/app/welcome/pages/reactive-form-errors/demo-reactive-form-errors-page.component.ts]: you can shorten this to `validators: PasswordValidator.areEqual`

File: showcase/src/app/welcome/pages/reactive-form-errors/password-validator.ts [hunk 30]
```diff
@@ -0,0 +1,32 @@
...
+		}
+
+		if (valid) {
+			/* tslint:disable-next-line:no-null-keyword*/
+			return null;
+		}
+
+		return {
+			areEqual: true
+		};
```
Comment (carlo-nomes) [inline showcase/src/app/welcome/pages/reactive-form-errors/password-validator.ts]: I prefer shorthands for this kind of simple logic, eg: `return valid ? null : { areEqual: true }`

File: showcase/src/app/welcome/welcome.module.ts [hunk 31]
```diff
@@ -4,15 +4,45 @@ import { GettingStartedPageComponent, HomePageComponent, NewsPageComponent, NoCo
...
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { MatInputModule } from "@angular/material/input";
 @NgModule({
 	imports: [
 		UIRouterModule.forChild({
 			states: NEWS_STATES
 		}),
-		SharedModule
+		SharedModule,
+		FormsModule,
```
Comment (carlo-nomes) [inline showcase/src/app/welcome/welcome.module.ts]: You should be able to remove FormsModule and ReactiveFormsModule here

**Comment by SuperITMan** [comment]
Superseded by #1721.
